osu.Game/Screens/Select/Details/AdvancedStats.cs

```diff
@@ -126,6 +126,22 @@ private void updateStatistics()
                     mod.ApplyToDifficulty(adjustedDifficulty);
             }
 
+            if (baseDifficulty != null && mods.Value.Any(m => m is ModRateAdjust))
+            {
+                adjustedDifficulty ??= new BeatmapDifficulty(baseDifficulty);
+
+                foreach (var mod in mods.Value.OfType<ModRateAdjust>())
+                {
+                    double speedChange = (float)mod.SpeedChange.Value;
+
+                    double preempt = (int)IBeatmapDifficultyInfo.DifficultyRange(adjustedDifficulty.ApproachRate, 1800, 1200, 450) / speedChange;
+                    adjustedDifficulty.ApproachRate = (float)(preempt > 1200 ? (1800 - preempt) / 120 : (1200 - preempt) / 150 + 5);
```

Comment:
not sure how to feel about this being hardcoded here like that. it's essentially specific to osu!-the-ruleset, catch _completely coincidentally_ uses the same formula. probably should be moved out to rulesets or something

Reply:
it might fail on custom rulesets so it should totally be moved out